verify.js

```diff
@@ -43,39 +38,9 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
     return done(new JsonWebTokenError('jwt must be a string'));
   }
 
-  var parts = jwtString.split('.');
-
-  if (parts.length !== 3){
-    return done(new JsonWebTokenError('jwt malformed'));
-  }
```

Comment:
I'd leave this assertion here, the rest of the code should not be executed if this error happens.

Reply:
No need for this as jws.decode already does the job.

Reply:
Fine. Can you add a test for this case? (since they are not failing I assume we don't have for this scenario currently)